src/parser/MetricCalculator.ts

```diff
@@ -45,42 +44,45 @@ export class MetricCalculator {
 
     /**
      * Calculates all basic single file metrics on the specified file.
-     * @param parseFile File for which the metric should be calculated.
+     * @param parsedFilePromise Promise for a parsed file for which the metric should be calculated.
      * @return A tuple that contains the path of the file and a Map
      * that relates each metric name to the calculated metric.
      */
-    async calculateMetrics(parseFile: ParseFile): Promise<[string, Map<string, MetricResult>]> {
-        try {
-            const metricResults = new Map<string, MetricResult>();
-            const resultPromises: Promise<MetricResult>[] = [];
+    async calculateMetrics(
+        parsedFilePromise: Promise<SimpleFile>
+    ): Promise<[string, Map<string, MetricResult>]> {
+        const parsedFile = await parsedFilePromise;
 
-            dlog(" ------------ Parsing File " + parseFile.filePath + ":  ------------ ");
+        if (!isParsedFile(parsedFile)) {
+            throw new Error(
+                "Unable to calculate file metrics because there was an error while creating the tree."
+            );
```

Comment:
Adapt error handling (see comments above).

Reply:
This error is catched by the error handling on which you commented above. So I expect it is fine?